controllers/platform/platform_detection.go

```diff
@@ -160,3 +162,16 @@ func SkipObjectStore(p configv1.PlatformType) bool {
 	}
 	return false
 }
+
+func IsPlatformROSAHCP() (bool, error) {
+	if platformInstance.platform == configv1.AWSPlatformType {
+		if platformInstance.infrastructure.Status.ControlPlaneTopology == configv1.ExternalTopologyMode {
+			for _, resourceTags := range platformInstance.infrastructure.Status.PlatformStatus.AWS.ResourceTags {
+				if resourceTags.Key == "red-hat-clustertype" && resourceTags.Value == "rosa" {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
```

Comment:
We should add a check for when the platformInstance is empty and return an error

Reply:
Changed logic in new commit